test/asn1_time_test.c

```diff
@@ -443,6 +443,26 @@ static int convert_asn1_to_time_t(int idx)
     return 1;
 }
 
+/*
+ * this test is here to exercise ossl_asn1_time_from_tm
+ * with an integer year close to INT_MAX.
+ */
+static int convert_tm_to_asn1_time(void)
+{
+  /* we need 64 bit time_t */
+#if ( (ULONG_MAX >> 31) >> 31 ) >= 1
+    time_t t;
+    ASN1_TIME* at;
+
+    TEST_int_ge(sizeof(time_t) * CHAR_BIT, 64);
+    t = 67768011791126057ULL;
+    at = ASN1_TIME_set(NULL, t);
```

Comment:
Should we sanity check the returned value is what we expect it to be?

Reply:
I don't think so, and wrote a code comment so the next person sees it.